app/src/main/java/com/example/sharingang/UserProfileViewModel.kt

```diff
@@ -1,20 +1,35 @@
 package com.example.sharingang
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
+import com.example.sharingang.items.Item
+import com.example.sharingang.items.ItemRepository
+import com.example.sharingang.items.ItemsViewModel
+import com.example.sharingang.users.CurrentUserProvider
 import com.example.sharingang.users.User
 import com.example.sharingang.users.UserRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class UserProfileViewModel @Inject constructor(
+    currentUserProvider: CurrentUserProvider,
+    private val itemRepository: ItemRepository,
     private val userRepository: UserRepository
 ) : ViewModel() {
 
-    private val _userId = MutableLiveData<String?>()
+    private val _userId = MutableLiveData<String?>(currentUserProvider.getCurrentUserId())
+    val userId: LiveData<String?>
+        get() = _userId
```

Comment:
We usually get the whole user instead of just the id with the val user, so no need to have a val userId here in my opinion.